wcfsetup/install/files/lib/system/endpoint/controller/core/labels/groups/ChangeLabelShowOrder.class.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace wcf\system\endpoint\controller\core\labels\groups;
+
+use Laminas\Diactoros\Response\JsonResponse;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use wcf\data\label\group\LabelGroup;
+use wcf\data\label\Label;
+use wcf\data\label\LabelEditor;
+use wcf\http\Helper;
+use wcf\system\endpoint\IController;
+use wcf\system\endpoint\PostRequest;
+use wcf\system\label\LabelHandler;
+use wcf\system\showOrder\ShowOrderHandler;
+use wcf\system\showOrder\ShowOrderItem;
+use wcf\system\WCF;
+
+/**
+ * API endpoint for changing the show order of the labels in a label group.
+ *
+ * @author      Marcel Werk
+ * @copyright   2001-2025 WoltLab GmbH
+ * @license     GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @since       6.2
+ */
+#[PostRequest('/core/labels/groups/{id:\d+}/label-show-order')]
```

Comment:
```suggestion
#[PostRequest('/core/labels/groups/{id:\d+}/labels/show-order')]
```

This represents a hierarchy in REST APIs, for example `GET /core/labels/groups/123/labels` would return the list of labels assigned to that group.

The location of the controller is fine though, it doesn’t have to mirror the structure 1:1, the naming is sufficient to keep it distinct.